### PR TITLE
gcp - dataproc clusters

### PR DESCRIPTION
**Pull request by evhenyshaliov**
Use case:

policies:
  - name: epam-gcp-dataproc_clusters_encrypted_with_cmek
    description: |
      Dataproc cluster is not encrypted with Customer-Managed Key
    resource: gcp.dataproc-clusters
    filters:
      - type: value
        key: config.encryptionConfig.gcePdKmsKeyName
        value: absent

### CODE REVIEW

File: tools/c7n_gcp/c7n_gcp/resources/compute.py [hunk 1]
```diff
@@ -617,3 +617,23 @@ class resource_type(TypeInfo):
         def get(client, resource_info):
             return client.execute_command(
                 'get', {'project': resource_info['project_id']})
+
+
+@resources.register('dataproc-clusters')
```
Comment (kapilt) [inline tools/c7n_gcp/c7n_gcp/resources/compute.py:622]: can we move this into a dataproc module, we generally segment service by module.

File: tools/c7n_gcp/c7n_gcp/resources/compute.py [hunk 2]
```diff
@@ -617,3 +617,23 @@ class resource_type(TypeInfo):
...
+        enum_spec = ('list', 'clusters[]', None)
+        scope_key = 'projectId'
+        name = id = 'name'
+        parent_spec = {
+            'resource': 'region',
+            'child_enum_params': {
+                ('name', 'region')},
+            'use_child_query': True,
+        }
+        default_report_fields = ['id', 'name', 'dnsName', 'creationTime', 'visibility']
```
Comment (kapilt) [inline tools/c7n_gcp/c7n_gcp/resources/compute.py:638]: we also have two additional required fields on metadata, one for asset inventory type, and one around urn construction. the later is a c7n specific construct that resembles an arn. I added support for both into the data fusion pr by way of example

File: tools/c7n_gcp/c7n_gcp/resources/compute.py (hunk 1 above)
Comment (evhenyshaliov) [reply tools/c7n_gcp/c7n_gcp/resources/compute.py]: done

File: tools/c7n_gcp/c7n_gcp/resources/compute.py (hunk 2 above)
Comment (evhenyshaliov) [reply tools/c7n_gcp/c7n_gcp/resources/compute.py]: done. Please check

File: tools/c7n_gcp/tests/test_dataproc.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+from gcp_common import BaseTest
+
+
+class TestDataprocCluster(BaseTest):
+
+    def test_query(self):
+        project_id = 'cloud-custodian'
+        factory = self.replay_flight_data('test_dataproc_clusters_query',
+                                          project_id=project_id)
+        p = self.load_policy(
```
Comment (kapilt) [inline tools/c7n_gcp/tests/test_dataproc.py:13]: we really need a testing helper here, to set the applicable regions to the ones desired to cut out the extraneous output recording. adding a mechanism for such to c7n_gcp/region.py:Region would be useful

**Comment by kapilt** [comment]
I've noticed in a few gcp resource prs, there's quite a bit of metadata that's not correct, id, name, reporting field. in future please add a test for get_urn() can help that, I've also added a field checker fixture that can assist, and updated this pr to use it.

File: tools/c7n_gcp/tests/test_dataproc.py (hunk 3 above)
Comment (kapilt) [reply tools/c7n_gcp/tests/test_dataproc.py]: I've gone ahead and a test fixture helper here for setting only required regions when using region as a parent, and removed the 20+ extraneous test files

**Review by kapilt — APPROVED** [review]
lgtm

**Pull request merged**